src/main/java/uk/gov/companieshouse/authcodenotification/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.companieshouse.authcodenotification.exception;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.owasp.encoder.Encode;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import uk.gov.companieshouse.authcodenotification.utils.ApiLogger;
+
+import java.util.HashMap;
+
+import static uk.gov.companieshouse.authcodenotification.utils.Constants.ERIC_REQUEST_ID_KEY;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @Value("${GLOBAL_EXCEPTION_HANDLER_TRUNCATE_LENGTH_CHARS:15000}")
+    private int truncationLength;
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<Object> handleException(Exception ex, WebRequest webRequest) {
+
+        var context = webRequest.getHeader(ERIC_REQUEST_ID_KEY);
+        var sanitisedExceptionMessage = truncate(Encode.forJava(ex.getMessage()));
+        var sanitisedStackTrace = truncate(Encode.forJava(ExceptionUtils.getStackTrace(ex)));
+        var sanitisedRootCause = truncate(Encode.forJava(ExceptionUtils.getStackTrace(ExceptionUtils.getRootCause(ex))));
+
+        HashMap<String, Object> logMap = new HashMap<>();
+        logMap.put("error", ex.getClass());
```

Comment:
i'd add the map constants into our constants class (error, stackTrace and rootCause)

Reply:
Even if only used in only one place (at the moment) - this class and this method?

To be honest though, I did wonder if these should be added to that structured logging builder, as the names we're using are quite general and they are now used in at least two places - here and in OE API. I'll plan to do that if you agree @markpit ?

Reply:
Yes, have done this now and also removed sanitisation, as discussed.